File: lib/aria/approach.mjs
// lib/aria/approach.mjs
// Module 9: Approach Strategy Generator
//
// Transforms a raw lead into an actionable package:
// - Draft opening message (tailored to market/language/culture)
// - Named contacts from contact DB
// - Ranked OEM list with export compliance status
// - Compliance checklist
// - Competitive positioning
//
// A lead with this package can be acted on the SAME AFTERNOON.
// Without it, 3 days of manual research before anyone picks up the phone.

import { getContactsByCountry, searchContacts } from './contacts.mjs';
import { searchKnowledge } from './knowledge.mjs';
import { queryLedger } from './intel_ledger.mjs';

// ── Market profiles ──────────────────────────────────────────────────────────
const MARKET_PROFILES = {
  'Angola':         { language: 'Portuguese', formality: 'HIGH', greeting: 'Exmo. Senhor', timezone: 'WAT (UTC+1)', currency: 'AOA/USD', approach: 'Formal letter via Defence Attaché or direct MoD. Portuguese essential. Reference CPLP framework. Personal meeting critical before commercial discussion.' },
  'Mozambique':     { language: 'Portuguese', formality: 'HIGH', greeting: 'Exmo. Senhor', timezone: 'CAT (UTC+2)', currency: 'MZN/USD', approach: 'Reference SADC framework and EU/SAMIM cooperation. Cabo Delgado situation creates urgency. Portuguese essential.' },
  'Guinea-Bissau':  { language: 'Portuguese', formality: 'MEDIUM', greeting: 'Exmo. Senhor', timezone: 'GMT', currency: 'XOF', approach: 'Small military — personal relationships paramount. ECOWAS/CPLP channels. Portuguese Creole awareness helpful.' },
  'Cape Verde':     { language: 'Portuguese', formality: 'MEDIUM', greeting: 'Exmo. Senhor', timezone: 'CVT (UTC-1)', currency: 'CVE', approach: 'Very small armed forces. Maritime security focus. Coast guard more active than military.' },
  'Nigeria':        { language: 'English', formality: 'HIGH', greeting: 'Dear Sir/Madam', timezone: 'WAT (UTC+1)', currency: 'NGN/USD', approach: 'Formal tender process. Local content requirements (DICON partnership). Multiple competing services (Army, Navy, Air Force) — each procures independently.' },
  'Kenya':          { language: 'English', formality: 'MEDIUM', greeting: 'Dear Sir/Madam', timezone: 'EAT (UTC+3)', currency: 'KES/USD', approach: 'Professional, competitive tender process. Reference AMISOM/ATMIS contribution. UK defence relationship historically strong.' },
  'Brazil':         { language: 'Portuguese', formality: 'HIGH', greeting: 'Exmo. Senhor', timezone: 'BRT (UTC-3)', currency: 'BRL/USD', approach: 'Large domestic industry (Embraer, Taurus, CBC). Offset/local content essential. Long procurement cycles. Portuguese native.' },
  'Indonesia':      { language: 'English/Indonesian', formality: 'HIGH', greeting: 'Yang Terhormat', timezone: 'WIB (UTC+7)', currency: 'IDR/USD', approach: 'MEF programme active. Offset essential. Korean/Turkish competitors strong. Formal MoD engagement.' },
  'Philippines':    { language: 'English', formality: 'MEDIUM', greeting: 'Dear Sir/Madam', timezone: 'PHT (UTC+8)', currency: 'PHP/USD', approach: 'Horizon 3 modernisation active. US FMS dominant but diversifying. English-speaking, relatively open process.' },
  'UAE':            { language: 'English/Arabic', formality: 'HIGH', greeting: 'Your Excellency', timezone: 'GST (UTC+4)', currency: 'AED/USD', approach: 'Wealthy buyer, sophisticated procurement. EDGE Group is local champion. Agent/representative essential. IDEX is key networking event.' },
  'Saudi Arabia':   { language: 'English/Arabic', formality: 'VERY HIGH', greeting: 'Your Royal Highness / Your Excellency', timezone: 'AST (UTC+3)', currency: 'SAR/USD', approach: 'Vision 2030 localisation. GAMI is procurement authority. Agent mandatory. Very long cycles (2-5 years).' },
};

// ── OEM ranking by product category ──────────────────────────────────────────
const OEM_BY_PRODUCT = {
  'armoured_vehicles': [
    { oem: 'Paramount Group', country: 'ZA', itar: false, price: 'MEDIUM', africa: 'STRONG', products: 'Mbombe, Marauder, Matador' },
    { oem: 'Otokar', country: 'TR', itar: false, price: 'LOW-MEDIUM', africa: 'GROWING', products: 'Cobra II, Arma 8x8' },
    { oem: 'FNSS', country: 'TR', itar: false, price: 'MEDIUM', africa: 'GROWING', products: 'Pars, Kaplan' },
    { oem: 'Rheinmetall', country: 'DE', itar: false, price: 'HIGH', africa: 'MODERATE', products: 'Lynx, Fuchs, Boxer' },
    { oem: 'Norinco', country: 'CN', itar: false, price: 'LOW', africa: 'STRONG', products: 'VN-1, VP-11, WMA301' },
  ],
  'uav_systems': [
    { oem: 'Baykar', country: 'TR', itar: false, price: 'MEDIUM', africa: 'STRONG', products: 'TB2, TB3, Akinci' },
    { oem: 'Turkish Aerospace (TUSAS)', country: 'TR', itar: false, price: 'MEDIUM', africa: 'GROWING', products: 'Anka, Aksungur' },
    { oem: 'Elbit Systems', country: 'IL', itar: false, price: 'HIGH', africa: 'MODERATE', products: 'Hermes 450/900, Skylark' },
    { oem: 'AVIC/CASC', country: 'CN', itar: false, price: 'LOW', africa: 'STRONG', products: 'Wing Loong I/II, CH-4/5' },
    { oem: 'General Atomics', country: 'US', itar: true, price: 'VERY HIGH', africa: 'LIMITED', products: 'MQ-9 Reaper, Gray Eagle' },
  ],
  'patrol_vessels': [
    { oem: 'Damen', country: 'NL', itar: false, price: 'MEDIUM', africa: 'STRONG', products: 'Stan Patrol, OPV 1800/2400' },
    { oem: 'Fincantieri', country: 'IT', itar: false, price: 'HIGH', africa: 'MODERATE', products: 'Corvettes, patrol boats' },
    { oem: 'Dearsan', country: 'TR', itar: false, price: 'LOW-MEDIUM', africa: 'GROWING', products: 'Patrol vessels, landing craft' },
    { oem: 'OCEA', country: 'FR', itar: false, price: 'MEDIUM', africa: 'STRONG', products: 'FPB, OPV — strong Francophone Africa' },
  ],
  'ammunition': [
    { oem: 'Nammo', country: 'NO', itar: false, price: 'MEDIUM', africa: 'MODERATE', products: 'All NATO calibres, 30mm, 155mm' },
    { oem: 'Rheinmetall', country: 'DE', itar: false, price: 'MEDIUM-HIGH', africa: 'MODERATE', products: '30mm, 35mm, 120mm, 155mm' },
    { oem: 'General Dynamics OTS', country: 'US', itar: true, price: 'HIGH', africa: 'LIMITED', products: 'All US mil-spec ammunition' },
    { oem: 'PMP (Denel)', country: 'ZA', itar: false, price: 'LOW-MEDIUM', africa: 'STRONG', products: 'Small arms, 12.7mm, 20mm' },
    { oem: 'CBC', country: 'BR', itar: false, price: 'LOW', africa: 'MODERATE', products: 'Small arms ammunition, Lusophone advantage' },
  ],
  'helicopters': [
    { oem: 'Airbus Helicopters', country: 'FR', itar: false, price: 'HIGH', africa: 'STRONG', products: 'H145M, H225M, Tiger' },
    { oem: 'Leonardo', country: 'IT', itar: false, price: 'HIGH', africa: 'STRONG', products: 'AW109, AW139, AW149' },
    { oem: 'Turkish Aerospace', country: 'TR', itar: false, price: 'MEDIUM', africa: 'GROWING', products: 'T129 Atak, T625 Gökbey' },
    { oem: 'Russian Helicopters', country: 'RU', itar: false, price: 'LOW', africa: 'LEGACY', products: 'Mi-17, Mi-35 — sanctioned, replacement opportunity' },
  ],
  'radar_air_defence': [
    { oem: 'Thales', country: 'FR', itar: false, price: 'HIGH', africa: 'STRONG', products: 'Ground Master, SHORAD' },
    { oem: 'SAAB', country: 'SE', itar: false, price: 'HIGH', africa: 'MODERATE', products: 'Giraffe, RBS-70' },
    { oem: 'Aselsan', country: 'TR', itar: false, price: 'MEDIUM', africa: 'GROWING', products: 'Korkut, Hisar, radar systems' },
    { oem: 'Rafael', country: 'IL', itar: false, price: 'VERY HIGH', africa: 'LIMITED', products: 'Iron Dome, David\'s Sling, Spyder' },
  ],
  'training': [
    { oem: 'BAE Systems', country: 'UK', itar: false, price: 'HIGH', africa: 'STRONG', products: 'Training academies, simulation' },
    { oem: 'L3Harris', country: 'US', itar: true, price: 'HIGH', africa: 'MODERATE', products: 'Flight simulation, ISR training' },
    { oem: 'Saab', country: 'SE', itar: false, price: 'MEDIUM-HIGH', africa: 'MODERATE', products: 'Tactical training, MILES' },
  ],
};

// ── Generate approach strategy ───────────────────────────────────────────────

/**
 * Generate a complete approach strategy for a lead.
 * Returns structured data that ARIA can present as actionable intelligence.
 */
export function generateApproach(market, product, context) {
  const profile = MARKET_PROFILES[market] || MARKET_PROFILES['Nigeria']; // default
  const contacts = getContactsByCountry(market);
  const productKey = mapProductToCategory(product);
  const oems = OEM_BY_PRODUCT[productKey] || [];

  // Filter OEMs by export feasibility
  const rankedOEMs = oems.map(o => ({
    ...o,
    exportRisk: o.itar ? 'HIGH — requires US DSP-5 licence' : 'LOW — non-ITAR, standard export licence',
    lusophoneAdvantage: ['BR', 'PT'].includes(o.country) && ['Portuguese'].includes(profile.language),
    africaFit: o.africa === 'STRONG' ? 3 : o.africa === 'GROWING' ? 2 : o.africa === 'MODERATE' ? 1 : 0,
  })).sort((a, b) => {
    // Prefer: non-ITAR > strong Africa presence > Lusophone advantage > lower price
    if (a.itar !== b.itar) return a.itar ? 1 : -1;
    if (a.africaFit !== b.africaFit) return b.africaFit - a.africaFit;
    return 0;
  });

  // Compliance checklist
  const compliance = [];
  compliance.push(profile.language === 'Portuguese' ? 'Lusophone market — Portuguese documentation required' : 'English documentation acceptable');
  if (oems.some(o => o.itar)) compliance.push('WARNING: Some OEM options are ITAR-controlled — verify US content before proceeding');
  compliance.push('End-user certificate required from ' + market + ' MoD');
  compliance.push('UK SIEL export licence application (20-60 working days)');
  compliance.push('Check UN Security Council embargo status for ' + market);
  compliance.push('Verify entity not on OFAC SDN, EU consolidated, or UK sanctions list');

  // Draft approach message
  const draftMessage = generateDraftMessage(market, product, profile, contacts);

  return {
    market,
    product: productKey,
    profile,
    contacts: contacts.slice(0, 5),
    rankedOEMs: rankedOEMs.slice(0, 4),
    compliance,
    draftMessage,
    timing: profile.approach,
    estimatedCycle: market === 'Saudi Arabia' ? '2-5 years' : market === 'UAE' ? '6-18 months' : '3-12 months',
  };
}

function generateDraftMessage(market, product, profile, contacts) {
  const authority = contacts.find(c => c.role === 'procurement_authority');
  const authorityName = authority ? authority.organisation : 'Ministry of Defence';

  if (profile.language === 'Portuguese') {
    return `Exmo. Senhor,\n\nTenho a honra de me dirigir a V. Exa. em nome da Arkmurus, empresa britânica especializada em assessoria de defesa e intermediação de equipamentos militares.\n\nTomamos conhecimento do interesse do ${authorityName} em ${product} e gostaríamos de apresentar soluções de fabricantes europeus e sul-africanos de referência, em total conformidade com as regulamentações internacionais de controlo de exportação.\n\nSolicito a oportunidade de apresentar as nossas credenciais e discutir como podemos apoiar os requisitos operacionais das Forças Armadas.\n\nAguardo a resposta de V. Exa.\nCom os melhores cumprimentos,\n[Nome]\nArkmurus Defence Advisory`;
  }

  return `Dear Sir/Madam,\n\nI write on behalf of Arkmurus, a UK-based defence advisory and brokering firm specialising in equipment procurement for armed forces across Africa and emerging markets.\n\nWe understand that the ${authorityName} of ${market} has requirements for ${product} and would welcome the opportunity to present solutions from leading European, South African, and Asian manufacturers, fully compliant with international export control regulations.\n\nWe would be grateful for the opportunity to present our credentials and discuss how we can support your operational requirements.\n\nYours faithfully,\n[Name]\nArkmurus Defence Advisory`;
}

function mapProductToCategory(product) {
  const p = (product || '').toLowerCase();
  if (p.match(/vehicle|armour|apc|ifv|mrap/)) return 'armoured_vehicles';
  if (p.match(/uav|drone|unmanned/)) return 'uav_systems';
  if (p.match(/patrol|vessel|coast|naval|ship|boat/)) return 'patrol_vessels';
  if (p.match(/ammun|round|calibr|bullet|shell|155|30mm|12\.7/)) return 'ammunition';
  if (p.match(/helicopter|rotorcraft|helo/)) return 'helicopters';
  if (p.match(/radar|air def|sam |shorad|manpad/)) return 'radar_air_defence';
  if (p.match(/train|simul|academy/)) return 'training';
  return 'armoured_vehicles'; // default
}

/**
 * Build approach context for ARIA prompt injection.
 */
export function getApproachContext(query) {
  const q = (query || '').toLowerCase();

  // Only inject if query relates to approaching/contacting/selling
  const triggers = ['approach', 'contact', 'how to', 'sell to', 'pitch', 'proposal', 'offer', 'engage', 'reach out', 'opening', 'introduce'];
  if (!triggers.some(t => q.includes(t))) return '';

  // Try to extract market and product from query
  const markets = Object.keys(MARKET_PROFILES);
  const market = markets.find(m => q.includes(m.toLowerCase()));
  if (!market) return '';

  const products = Object.keys(OEM_BY_PRODUCT);
  const product = products.find(p => q.includes(p.replace(/_/g, ' '))) || '';

  const strategy = generateApproach(market, product, q);

  return '\n\n[APPROACH STRATEGY — auto-generated for ' + market + ']\n' +
    'Language: ' + strategy.profile.language + ' | Formality: ' + strategy.profile.formality + '\n' +
    'Greeting: ' + strategy.profile.greeting + '\n' +
    'Timezone: ' + strategy.profile.timezone + '\n' +
    'Approach: ' + strategy.profile.approach + '\n' +
    'Estimated cycle: ' + strategy.estimatedCycle + '\n' +
    (strategy.contacts.length ? 'Key contacts: ' + strategy.contacts.map(c => c.name + ' (' + c.title + ')').join('; ') + '\n' : '') +
    (strategy.rankedOEMs.length ? 'Ranked OEMs: ' + strategy.rankedOEMs.map(o => o.oem + ' [' + o.country + '] ' + o.price + ' ' + (o.itar ? 'ITAR' : 'non-ITAR')).join('; ') + '\n' : '') +
    'Compliance: ' + strategy.compliance.join(' | ');
}

File: aria_service/intel/approach.py
"""
Market Approach Strategy Generator.
Ported from lib/aria/approach.mjs.
"""
from __future__ import annotations

import re
from typing import Any

from .contacts import search_contacts

# ── Market Profiles ──────────────────────────────────────────────────────────

MARKET_PROFILES: dict[str, dict] = {
    "Angola": {"language": "Portuguese", "formality": "HIGH", "greeting": "Exmo. Senhor", "tz": "WAT (UTC+1)", "currency": "AOA/USD", "notes": "Formal letter via Defence Attaché, CPLP framework", "cycle": "3-12 months"},
    "Mozambique": {"language": "Portuguese", "formality": "HIGH", "greeting": "Exmo. Senhor", "tz": "CAT (UTC+2)", "currency": "MZN/USD", "notes": "SADC/EU SAMIM cooperation, Cabo Delgado urgency", "cycle": "3-12 months"},
    "Nigeria": {"language": "English", "formality": "HIGH", "greeting": "Dear Sir/Madam", "tz": "WAT (UTC+1)", "currency": "NGN/USD", "notes": "Formal tender, local content (DICON partnership)", "cycle": "3-12 months"},
    "Kenya": {"language": "English", "formality": "MEDIUM", "greeting": "Dear Sir/Madam", "tz": "EAT (UTC+3)", "currency": "KES/USD", "notes": "Competitive tender, AMISOM/ATMIS reference", "cycle": "3-12 months"},
    "Saudi Arabia": {"language": "English/Arabic", "formality": "VERY HIGH", "greeting": "Your Royal Highness", "tz": "AST (UTC+3)", "currency": "SAR/USD", "notes": "Vision 2030, GAMI is gatekeeper, agent mandatory, 50-60% offset", "cycle": "2-5 years"},
    "UAE": {"language": "English/Arabic", "formality": "HIGH", "greeting": "Your Excellency", "tz": "GST (UTC+4)", "currency": "AED/USD", "notes": "EDGE Group partnership, agent essential, IDEX meeting point, 60% Tawazun", "cycle": "6-18 months"},
    "Senegal": {"language": "French", "formality": "HIGH", "greeting": "Excellence", "tz": "GMT", "currency": "XOF", "notes": "Nexter/Airbus historical preference, ECOWAS contributor", "cycle": "6-18 months"},
    "Indonesia": {"language": "English/Indonesian", "formality": "HIGH", "greeting": "Yang Terhormat", "tz": "WIB (UTC+7)", "currency": "IDR/USD", "notes": "MEF programme, 35-85% offset mandatory, Korean competitors strong", "cycle": "12-24 months"},
    "Philippines": {"language": "English", "formality": "MEDIUM", "greeting": "Dear Sir/Madam", "tz": "PHT (UTC+8)", "currency": "PHP/USD", "notes": "Horizon 3 modernisation, US FMS dominant, diversifying", "cycle": "6-18 months"},
    "Brazil": {"language": "Portuguese", "formality": "HIGH", "greeting": "Exmo. Senhor", "tz": "BRT (UTC-3)", "currency": "BRL/USD", "notes": "Embraer domestic champion, 100% offset >R$50M", "cycle": "6-18 months"},
    "Ghana": {"language": "English", "formality": "MEDIUM", "greeting": "Dear Sir/Madam", "tz": "GMT", "currency": "GHS/USD", "notes": "Competitive tender, ECOWAS contributor", "cycle": "6-12 months"},
    "Ethiopia": {"language": "English", "formality": "HIGH", "greeting": "Your Excellency", "tz": "EAT (UTC+3)", "currency": "ETB/USD", "notes": "Post-conflict reconstruction, relationship-first", "cycle": "6-18 months"},
    "Guinea-Bissau": {"language": "Portuguese", "formality": "HIGH", "greeting": "Exmo. Senhor", "tz": "GMT", "currency": "XOF", "notes": "ECOWAS donor-dependent, Arkmurus family connections", "cycle": "6-18 months"},
    "Cape Verde": {"language": "Portuguese", "formality": "HIGH", "greeting": "Exmo. Senhor", "tz": "CVT (UTC-1)", "currency": "CVE/USD", "notes": "Small force, maritime focus, CPLP framework", "cycle": "6-18 months"},
}

# ── OEM Rankings by Product ──────────────────────────────────────────────────

OEM_RANKINGS: dict[str, list[dict]] = {
    "armoured_vehicles": [
        {"oem": "Paramount Group", "country": "ZA", "itar": False, "price": "MEDIUM", "africa": "STRONG"},
        {"oem": "Otokar", "country": "TR", "itar": False, "price": "LOW-MEDIUM", "africa": "GROWING"},
        {"oem": "FNSS", "country": "TR", "itar": False, "price": "MEDIUM", "africa": "GROWING"},
        {"oem": "Rheinmetall", "country": "DE", "itar": False, "price": "HIGH", "africa": "MODERATE"},
        {"oem": "Norinco", "country": "CN", "itar": False, "price": "LOW", "africa": "STRONG"},
    ],
    "uav_systems": [
        {"oem": "Baykar", "country": "TR", "itar": False, "price": "MEDIUM", "africa": "STRONG"},
        {"oem": "Elbit Systems", "country": "IL", "itar": False, "price": "HIGH", "africa": "STRONG"},
        {"oem": "Turkish Aerospace", "country": "TR", "itar": False, "price": "MEDIUM", "africa": "GROWING"},
        {"oem": "AVIC/CATIC", "country": "CN", "itar": False, "price": "LOW", "africa": "GROWING"},
        {"oem": "IAI", "country": "IL", "itar": False, "price": "HIGH", "africa": "MODERATE"},
    ],
    "patrol_vessels": [
        {"oem": "Damen", "country": "NL", "itar": False, "price": "MEDIUM", "africa": "STRONG"},
        {"oem": "Navantia", "country": "ES", "itar": False, "price": "MEDIUM", "africa": "MODERATE"},
        {"oem": "Fincantieri", "country": "IT", "itar": False, "price": "HIGH", "africa": "MODERATE"},
    ],
    "ammunition": [
        {"oem": "Rheinmetall Denel", "country": "ZA/DE", "itar": False, "price": "MEDIUM", "africa": "STRONG"},
        {"oem": "Norinco", "country": "CN", "itar": False, "price": "LOW", "africa": "STRONG"},
        {"oem": "General Dynamics OTS", "country": "US", "itar": True, "price": "HIGH", "africa": "LIMITED"},
    ],
    "helicopters": [
        {"oem": "Leonardo", "country": "IT", "itar": False, "price": "HIGH", "africa": "STRONG"},
        {"oem": "Airbus Helicopters", "country": "FR", "itar": False, "price": "HIGH", "africa": "MODERATE"},
        {"oem": "Turkish Aerospace", "country": "TR", "itar": False, "price": "MEDIUM", "africa": "GROWING"},
    ],
    "radar_air_defence": [
        {"oem": "Aselsan", "country": "TR", "itar": False, "price": "MEDIUM", "africa": "GROWING"},
        {"oem": "Thales", "country": "FR", "itar": False, "price": "HIGH", "africa": "MODERATE"},
        {"oem": "Rafael", "country": "IL", "itar": False, "price": "HIGH", "africa": "MODERATE"},
        {"oem": "Saab", "country": "SE", "itar": False, "price": "HIGH", "africa": "LIMITED"},
    ],
    "training": [
        {"oem": "Leonardo (M-346)", "country": "IT", "itar": False, "price": "HIGH", "africa": "STRONG"},
        {"oem": "KAI (FA-50)", "country": "KR", "itar": False, "price": "MEDIUM", "africa": "GROWING"},
        {"oem": "Embraer (A-29)", "country": "BR", "itar": False, "price": "MEDIUM", "africa": "MODERATE"},
    ],
}


def _map_product_to_category(product: str) -> str:
    pl = product.lower()
    mapping = [
        ("armoured_vehicles", ["armour", "vehicle", "apc", "ifv", "mrap", "tank"]),
        ("uav_systems", ["uav", "drone", "unmanned", "uas", "rpas"]),
        ("patrol_vessels", ["vessel", "patrol", "opv", "corvette", "frigate", "naval"]),
        ("ammunition", ["ammunit", "ammo", "round", "mortar", "shell", "calibre"]),
        ("helicopters", ["helicopter", "rotary", "helo"]),
        ("radar_air_defence", ["radar", "air defen", "sam", "shorad", "c-ram"]),
        ("training", ["training", "trainer", "simulation", "exercise"]),
    ]
    for cat, kws in mapping:
        if any(k in pl for k in kws):
            return cat
    return ""


# ── Draft Messages ───────────────────────────────────────────────────────────

def _draft_message(profile: dict, product: str, market: str) -> str:
    lang = profile.get("language", "English")
    greeting = profile.get("greeting", "Dear Sir/Madam")

    if "Portuguese" in lang:
        return (
            f"{greeting},\n\n"
            f"A Arkmurus Group apresenta os seus cumprimentos e manifesta o seu interesse "
            f"em apoiar as necessidades de {product or 'defesa'} de {market}. "
            f"Enquanto consultora britânica de defesa com forte presença no espaço CPLP, "
            f"temos experiência comprovada na facilitação de aquisições de defesa em conformidade "
            f"com os regulamentos internacionais aplicáveis.\n\n"
            f"Gostaríamos de agendar uma reunião para discutir como podemos apoiar os vossos objectivos.\n\n"
            f"Com os melhores cumprimentos,\nArkmurus Group"
        )
    elif "French" in lang:
        return (
            f"{greeting},\n\n"
            f"Arkmurus Group vous présente ses salutations et exprime son intérêt à soutenir "
            f"les besoins de défense de {market}. En tant que cabinet-conseil britannique en défense, "
            f"nous disposons d'une expertise éprouvée en facilitation d'acquisitions de défense "
            f"conformément aux réglementations internationales.\n\n"
            f"Nous souhaiterions convenir d'une réunion pour discuter de notre collaboration.\n\n"
            f"Avec nos salutations distinguées,\nArkmurus Group"
        )
    elif "Arabic" in lang:
        return (
            f"{greeting},\n\n"
            f"Arkmurus Group presents its compliments and expresses interest in supporting "
            f"the defence requirements of {market}. As a UK-based defence advisory with "
            f"extensive regional experience and compliance-first approach, we bring both "
            f"established OEM relationships and deep understanding of Middle East procurement.\n\n"
            f"We would welcome the opportunity to discuss how we may be of service.\n\n"
            f"With highest regards,\nArkmurus Group"
        )
    else:
        return (
            f"{greeting},\n\n"
            f"Arkmurus Group presents its compliments and expresses interest in supporting "
            f"the {product or 'defence'} requirements of {market}. As a UK-registered defence "
            f"advisory and brokerage firm, we bring OEM relationships, export compliance "
            f"expertise, and proven experience across African and emerging markets.\n\n"
            f"We would welcome the opportunity to discuss collaboration.\n\n"
            f"With best regards,\nArkmurus Group"
        )


# ── Public API ───────────────────────────────────────────────────────────────

def generate_approach(market: str, product: str = "", context: str = "") -> dict:
    """Generate a complete approach strategy for a market/product."""
    profile = MARKET_PROFILES.get(market)
    if not profile:
        # Fuzzy match
        ml = market.lower()
        for k, v in MARKET_PROFILES.items():
            if ml in k.lower():
                profile = v
                market = k
                break
    if not profile:
        return {"error": f"No profile for market: {market}"}

    category = _map_product_to_category(product) if product else ""
    ranked_oems = OEM_RANKINGS.get(category, [])

    contacts = search_contacts(market)[:4]

    compliance = [
        f"Language: {profile['language']} required for all formal correspondence",
        "ITAR check: " + ("⚠️ ITAR-controlled items require US State Dept approval" if any(o.get("itar") for o in ranked_oems) else "Non-ITAR OEMs preferred"),
        f"End-user certificate required from {market} MoD",
        "UK SIEL export licence (20-60 working days processing)",
        f"UN Security Council embargo check for {market}",
        "OFAC SDN / EU consolidated / UK sanctions list verification",
    ]

    return {
        "market": market,
        "product": product or "general",
        "profile": profile,
        "contacts": contacts,
        "rankedOEMs": ranked_oems,
        "compliance": compliance,
        "draftMessage": _draft_message(profile, product, market),
        "timing": profile.get("notes", ""),
        "estimatedCycle": profile.get("cycle", "3-12 months"),
    }


def get_approach_context(query: str) -> str:
    """Synchronous prompt injection — triggers only on approach-related queries."""
    ql = query.lower()
    approach_kws = ["approach", "strategy", "enter", "engage", "pitch", "message", "draft", "contact"]
    if not any(k in ql for k in approach_kws):
        return ""
    # Find market in query
    for market in MARKET_PROFILES:
        if market.lower() in ql:
            result = generate_approach(market)
            if "error" in result:
                return ""
            p = result["profile"]
            lines = [
                f"\n[APPROACH STRATEGY — {market}]",
                f"Language: {p['language']} | Formality: {p['formality']} | Greeting: {p['greeting']}",
                f"Notes: {p['notes']}",
                f"Cycle: {result['estimatedCycle']}",
            ]
            if result["rankedOEMs"]:
                lines.append("Ranked OEMs: " + ", ".join(f"{o['oem']} ({o['country']})" for o in result["rankedOEMs"][:4]))
            return "\n".join(lines)
    return ""
